lib/DDG/Spice/Twitter.pm

```diff
@@ -5,16 +5,22 @@ use strict;
 use DDG::Spice;
 
 spice to => 'https://duckduckgo.com/tw.js?user=$1&callback={{callback}}&current=1';
-triggers query => qr/^(?:twitter\s)?@([a-z0-9_]+)$|^twitter\s([a-z0-9_]+)$/i;
+spice proxy_cache_valid => '200 1h';
+
+my $grammar = qr/on|at/;
+my $other = qr/user|account|id/;
+
+triggers query_lc => qr/^(?:twitter\s)?(?:$other\s)?@([a-z0-9_]+)$/i;
+triggers query_lc => qr/^twitter\s(?:$other\s)?([a-z0-9_]+)$/i;
+triggers query_lc => qr/^([a-z0-9_]+)(?:'s)?\s(?:$other\s)?(?:$grammar\s)?(?:twitter)(?:\s$other)?$/i;
+triggers query_lc => qr/^@([a-z0-9_]+)(?:'s)?\s(?:$other\s)?(?:$grammar\s)?twitter?(?:\s$other)?$/i;
```

Comment:
@nalinbhardwaj after testing this, I think these triggers are too broad. Let's stick with the existing triggers. A few weeks back I made changes to the special internal triggering of the Twitter IA which should cover most of these kinds of queries.

The API endpoint we use is designed to lookup a username, but some of these queries suggest they're trying to find a person/company twitter account which will lead to odd results.

Also this will trigger on queries like "delete twitter account", "create twitter account".